cub/block/block_radix_sort.cuh

```diff
@@ -522,6 +574,135 @@ public:
         SortBlocked(keys, values, begin_bit, end_bit, Int2Type<false>(), Int2Type<KEYS_ONLY>());
     }
 
+    //! @rst
+    //! Performs an ascending block-wide radix sort over a 
+    //! :ref:`blocked arrangement <flexible-data-arrangement>` of keys.
+    //!
+    //! * @granularity
+    //! * @smemreuse
+    //!
+    //! Snippet
+    //! ==========================================================================
+    //!
+    //! Let's consider a user-defined ``custom_t`` type below. To sort an array of
+    //! ``custom_t`` objects, we have to tell CUB about relevant members of the
+    //! ``custom_t`` type. We do this by providing a decomposer that returns a
+    //! tuple of references to relevant members of the key.
+    //!
+    //! .. literalinclude:: ../../test/catch2_test_block_radix_sort_custom.cu
+    //!     :language: c++
+    //!     :dedent:
+    //!     :start-after: example-begin custom-type
+    //!     :end-before: example-end custom-type
+    //!
+    //! The code snippet below illustrates a sort of 2 keys that
+    //! are partitioned in a :ref:`blocked arrangement <flexible-data-arrangement>` across 2 threads
+    //! where each thread owns 1 key.
+    //!
+    //! .. literalinclude:: ../../test/catch2_test_block_radix_sort_custom.cu
+    //!     :language: c++
+    //!     :dedent:
+    //!     :start-after: example-begin keys-bits
+    //!     :end-before: example-end keys-bits
+    //!
+    //! @endrst
+    //!
+    //! @tparam DecomposerT
+    //!   **[inferred]** Type of a callable object responsible for decomposing a 
+    //!   ``KeyT`` into a tuple of references to its constituent arithmetic types:
+    //!   ``::cuda::std::tuple<ArithmeticTs&...> operator()(KeyT &key)``. 
+    //!   The leftmost element of the tuple is considered the most significant. 
+    //!   The call operator must not modify members of the key.
+    //!
+    //! @param[in,out] keys 
+    //!   Keys to sort
+    //!
+    //! @param decomposer
+    //!   Callable object responsible for decomposing a ``KeyT`` into a tuple of
+    //!   references to its constituent arithmetic types. The leftmost element of 
+    //!   the tuple is considered the most significant. The call operator must not 
+    //!   modify members of the key.
+    //!
+    //! @param[in] begin_bit 
+    //!   The least-significant bit index (inclusive) needed for 
+    //!   key comparison
+    //!
+    //! @param[in] end_bit 
+    //!   The most-significant bit index (exclusive) needed for key 
+    //!   comparison (e.g., `(sizeof(float) + sizeof(long long int)) * 8`)
+    template <class DecomposerT>
+    __device__ __forceinline__         //
+      typename ::cuda::std::enable_if< //
+        !::cuda::std::is_convertible<DecomposerT, int>::value>::type
```

Comment:
```suggestion
      ::cuda::std::__enable_if_t<!::cuda::std::is_convertible<DecomposerT, int>::value>>
```

Feel free to disregard if you want to avoid `__ugly` constructs